### PR TITLE
Rewrite the prstate action

### PR DESCRIPTION
**Pull request by npmccallum**
There were numerous problems with the previous version. We hopefully simplify
this by merging all events into a single stream and then processing them in
reverse order. This allows us to have later events modify how earlier events
are processed. The latest event that assigns responsibility, possibly in view
of later events, is the effective event.

This is a difficult problem and should probably be rewritten in GraphQL at some
point. But this is probably good enough for now.

Fixes: #25 

Co-authored-by: Mark Bestavros <mbestavr@redhat.com>

### CODE REVIEW

**Comment by npmccallum** [comment]
This PR replaces #31.

File: enarx-pr-assign [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: Apache-2.0
+
+import enarxbot
+import datetime
+
+def merge_events_and_reviews(pr):
+    "Gets a summary of all events on a PR by merging isssue events and reviews."
```
Comment (mbestavros) [inline enarx-pr-assign:8]: `isssue` -> `issue`

Comment (npmccallum) [reply enarx-pr-assign]: Fixed.

**Comment by mbestavros** [comment]
@npmccallum #25 is still not linked. You have to edit the PR description for Github to link issues.

**Pull request merged**